sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -121,8 +123,13 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
-    System.getProperty("java.io.tmpdir") + File.separator + db + ".db"
+  def getDatabasePath(dbName: String, path: Option[String]): String = {
+    val dbPath =
+      path.map(new Path(_))
+        .getOrElse(
+          new Path(new Path(System.getProperty("java.io.tmpdir")), dbName.toLowerCase() + ".db"))
+    val dbLocation = dbPath.toString
+    if (dbLocation.endsWith(File.separator)) dbLocation.dropRight(1) else dbLocation
```

Comment:
actually, @gatorsmile why do we need to change the `createDatabasePath` here for `SessionCatalog`? What's wrong with the old implementation?